sys/dev/hid/hidraw.c

```diff
@@ -527,6 +579,32 @@ hidraw_ioctl(struct cdev *dev, u_long cmd, caddr_t addr, int flag,
 	if (sc == NULL)
 		return (EIO);
 
+#ifdef COMPAT_FREEBSD32
+	hgd = (struct hidraw_gen_descriptor *)addr;
+	switch (cmd) {
+	case HIDRAW_GET_REPORT_DESC32:
+	case HIDRAW_GET_REPORT32:
+	case HIDRAW_SET_REPORT_DESC32:
+	case HIDRAW_SET_REPORT32:
+		cmd = _IOC_NEWTYPE(cmd, struct hidraw_gen_descriptor);
+		hgd32 = (struct hidraw_gen_descriptor32 *)addr;
+		hgd = &local_hgd;
+		PTRIN_CP(*hgd32, *hgd, hgd_data);
+		CP(*hgd32, *hgd, hgd_lang_id);
+		CP(*hgd32, *hgd, hgd_maxlen);
+		CP(*hgd32, *hgd, hgd_actlen);
+		CP(*hgd32, *hgd, hgd_offset);
+		CP(*hgd32, *hgd, hgd_config_index);
+		CP(*hgd32, *hgd, hgd_string_index);
+		CP(*hgd32, *hgd, hgd_iface_index);
+		CP(*hgd32, *hgd, hgd_altif_index);
+		CP(*hgd32, *hgd, hgd_endpt_index);
+		CP(*hgd32, *hgd, hgd_report_type);
+		/* Don't copy reserved */
+		break;
```

Comment:
Style: Should maybe add:

default:
break;

at the end there.

Reply:
Definitely not. Maybe:
```
default:
panic(...);
```
but not a default case that does nothing for a programming error.

Reply:
Sorry, in this case one would be fine, but IMO utterly pointless. This is your code though so if you want pointless default cases I can add them.

Reply:
OK, as long as the compiler doesn't complain.